firedrake/adjoint/dirichletbc.py

```diff
@@ -0,0 +1,42 @@
+from pyadjoint.overloaded_type import FloatingType
+from .blocks import DirichletBCBlock
+from pyadjoint.tape import no_annotations
+
+
+class DirichletBCMixin(FloatingType):
+    @staticmethod
+    def _ad_annotate_init(init):
+        def wrapper(self, *args, **kwargs):
+            FloatingType.__init__(self,
+                                  *args,
+                                  block_class=DirichletBCBlock,
+                                  _ad_args=args,
+                                  _ad_floating_active=True,
+                                  **kwargs)
+            init(self, *args, **kwargs)
```

Comment:
This is the concrete `__init__` right? That does stuff with projection for some dirichlet bc calls. Should those be annotated (as, it seems, is happening here?)

Reply:
I think it's at least not harmful to annotate the projections.